### PR TITLE
Users and groups models

### PR DESCRIPTION
**Pull request by Azrael-git**
Done task about models

### CODE REVIEW

File: apps/accounts/migrations/0001_initial.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
+# Generated by Django 2.2.3 on 2019-07-21 09:33
```
Comment (klekhaav) [inline apps/accounts/migrations/0001_initial.py]: Please add all files with migrations to .gitignore

File: apps/accounts/models.py [hunk 2]
```diff
@@ -1,4 +1,49 @@
 from django.db import models
 
+# Base user class
+class User(models.Model):
```
Comment (klekhaav) [inline apps/accounts/models.py]: User must be from abstract user class or basic user class

File: apps/accounts/models.py [hunk 3]
```diff
@@ -1,4 +1,49 @@
...
+    surname = models.CharField(max_length=64, blank=False)          # Ivanov
+    patronymic = models.CharField(max_length=64, blank=False)       # Ivanovych
+    email = models.EmailField(max_length=64, blank=False)           # ivanov@gmail.com
+    password = models.CharField(max_length=64, blank=False)         # 21kjfs23iuk
+    avatar = models.ImageField(upload_to='static/images/', blank=True, max_length=1000)     # select image
 
-# Create your models here.
+    # Returns object's name
+    def __str__(self):
+        return(self.name)
```
Comment (klekhaav) [inline apps/accounts/models.py]: method to return full user name or nickname (selected by user) do not use sensitive data to return

File: apps/accounts/models.py [hunk 4]
```diff
@@ -1,4 +1,49 @@
...
+
+    # Helps in class inheritance
+    class Meta:
+        abstract = True
```
Comment (klekhaav) [inline apps/accounts/models.py]: Final user model cannot be abstract

File: apps/accounts/models.py [hunk 5]
```diff
@@ -1,4 +1,49 @@
...
+
+# Class for groups
+class Group(models.Model):
+    amount = models.IntegerField(blank=False)                                               # 25
```
Comment (klekhaav) [inline apps/accounts/models.py]: autofilled field

File: apps/accounts/models.py [hunk 6]
```diff
@@ -1,4 +1,49 @@
...
+    group_code = models.CharField(max_length=5, blank=False)                                # CB-41
+    
+    # Using 'Student' and 'Teacher' because of cyclic keys
+    studens = models.ForeignKey('Student',
```
Comment (klekhaav) [inline apps/accounts/models.py]: add group name by choice field

File: apps/accounts/models.py [hunk 7]
```diff
@@ -1,4 +1,49 @@
...
+        return(self.name)
+
+    # Helps in class inheritance
+    class Meta:
+        abstract = True
+
+# Class for groups
+class Group(models.Model):
+    amount = models.IntegerField(blank=False)                                               # 25
+    group_code = models.CharField(max_length=5, blank=False)                                # CB-41
```
Comment (klekhaav) [inline apps/accounts/models.py]: related field to group id

File: apps/accounts/models.py [hunk 8]
```diff
@@ -1,4 +1,49 @@
...
+    curator = models.OneToOneField('Teacher', on_delete=models.CASCADE, primary_key=True)   # Kotsuba A. U
+
+    # Returns group code (CB-41 for example)
+    def __str__(self):
+        return(self.group_code)
+
+# Class for students
+class Student(User):
+    record_book_code = models.CharField(max_length=8)                               # requires clarification (smth. like 182.10)
+    group = models.OneToOneField(Group,
```
Comment (klekhaav) [inline apps/accounts/models.py]: one way relation please

File: apps/accounts/models.py [hunk 9]
```diff
@@ -1,4 +1,49 @@
...
+
+# Class for students
+class Student(User):
+    record_book_code = models.CharField(max_length=8)                               # requires clarification (smth. like 182.10)
+    group = models.OneToOneField(Group, 
+            on_delete=models.CASCADE, 
+            primary_key=True,
+            blank=True)                                                             # CB-41    
+    language = models.CharField(max_length=24)                                      # Ukrainian
+    faculty = models.CharField(max_length=64)                                       # FKNIT
```
Comment (klekhaav) [inline apps/accounts/models.py]: define default value

File: apps/accounts/models.py [hunk 10]
```diff
@@ -1,4 +1,49 @@
...
+    pulpit = models.CharField(max_length=128)                                       # Cathedra of computer systems
```
Comment (klekhaav) [inline apps/accounts/models.py]: define default value

File: apps/accounts/models.py [hunk 11]
```diff
@@ -1,4 +1,49 @@
...
+            on_delete=models.CASCADE, 
+            related_name="%(app_label)s_%(class)s_related", blank=True)                     # [Ivanov, Petrov, Vasilenko, ...]
+    curator = models.OneToOneField('Teacher', on_delete=models.CASCADE, primary_key=True)   # Kotsuba A. U
+
+    # Returns group code (CB-41 for example)
+    def __str__(self):
+        return(self.group_code)
+
+# Class for students
+class Student(User):
```
Comment (klekhaav) [inline apps/accounts/models.py]: Define as group, select group in user model

File: apps/accounts/models.py [hunk 12]
```diff
@@ -1,4 +1,49 @@
...
+    group = models.OneToOneField(Group, 
+            on_delete=models.CASCADE, 
+            primary_key=True,
+            blank=True)                                                             # CB-41    
+    language = models.CharField(max_length=24)                                      # Ukrainian
+    faculty = models.CharField(max_length=64)                                       # FKNIT
+    pulpit = models.CharField(max_length=128)                                       # Cathedra of computer systems
+
+# Class for teachers
+class Teacher(User):
```
Comment (klekhaav) [inline apps/accounts/models.py]: Define as group, select group in user model

File: apps/post/apps.py [hunk 13]
```diff
@@ -0,0 +1,5 @@
+from django.apps import AppConfig
+
+
+class PostConfig(AppConfig):
+    name = 'post'
```
Comment (klekhaav) [inline apps/post/apps.py]: Post? You need to define user admin model

File: apps/post/admin.py [hunk 14]
```diff
@@ -0,0 +1,4 @@
+from django.contrib import admin
+from .models import Post
+
+admin.site.register(Post)
```
Comment (klekhaav) [inline apps/post/admin.py]: Post? You need to define user admin model

File: apps/post/migrations/0001_initial.py [hunk 15]
```diff
@@ -0,0 +1,24 @@
+# Generated by Django 2.1 on 2019-08-01 19:37
```
Comment (klekhaav) [inline apps/post/migrations/0001_initial.py]: ignore migrations

File: apps/post/models.py [hunk 16]
```diff
@@ -0,0 +1,11 @@
+from django.db import models
+
+class Post(models.Model):
```
Comment (klekhaav) [inline apps/post/models.py]: Remove Posts model we dont need it yet

File: settings/base.py [hunk 17]
```diff
@@ -39,6 +39,7 @@
     'accounts',
     'djangotoolbox',
     'pages',
+    'post',
```
Comment (klekhaav) [inline settings/base.py]: Remove Posts app we dont need it yet

**Review by klekhaav — CHANGES_REQUESTED** [review]

File: apps/accounts/models.py (hunk 7 above)
Comment (IgnisAvis) [reply apps/accounts/models.py]: > related field to group id

What this means? We use this field to save how much students in AcademicGroup

File: apps/accounts/models.py (hunk 5 above)
Comment (Azrael-git) [reply apps/accounts/models.py]: > autofilled field

What this means? We use this field to save how much students in AcademicGroup

**Pull request closed without merging**